source/AI.h

```diff
@@ -175,6 +176,15 @@ template <class Type>
 	void IssueOrders(const PlayerInfo &player, const Orders &newOrders, const std::string &description);
 	// Convert order types based on fulfillment status.
 	void UpdateOrders(const Ship &ship);
+	// Get the max range of a list of weapons (usually a ship's weapons) excluding any AMs
+	double GetMaxRange(const std::vector<Hardpoint> &weapons) const;
+	// Overload and add the option to specificy forward facing guns or turrets only
+	double GetMaxRange(const std::vector<Hardpoint> &weapons, bool aimable) const;
+	// Get the min range of a list of weapons (usually a ship's weapons) excluding any AMs
+	double GetMinRange(const std::vector<Hardpoint> &weapons) const;
+	// Overload and add the option to specificy forward facing guns or turrets only
+	double GetMinRange(const std::vector<Hardpoint> &weapons, bool aimable) const;
+
```

Comment:
These are unused, so they should be deleted.

If they need to be kept, they do not need to be class methods (they need no information or methods from the AI class).

Reply:
You're right, they don't belong here anymore.  Where would you put them if not in AI?  I can't see any general purpose classes, but maybe I missed one.

Reply:
They'd fit as free functions, but they don't need to be added at all at the current point in time.

Reply:
They could be useful because their job is already done several times in the code besides where I was going to use it.  There is a drawback to using them though: if you make them atomic like I did, you can't grab both min and max at once and you need to iterate through the list 2x.  There could be a 3rd set of functions to fix that but then you need to either pass by reference (not the worst thing) or return an array (or the correct term in C++, I don't think it's array).